plugins/chat/app/models/chat/tracking_state_report.rb

```diff
@@ -8,19 +8,24 @@ class TrackingStateReport
     attr_accessor :channel_tracking, :thread_tracking
 
     class TrackingStateInfo
-      attr_accessor :unread_count, :mention_count
+      attr_accessor :unread_count, :mention_count, :last_reply_created_at
```

Comment:
One thing which bothers me here is that we have this on channel:

```
#  last_message_sent_at        :datetime
```

So I have multiple questions:
- should we have the same on thread model and use it instead?
- should we remove this and just use a tracking state report when we need it in a serializer?

Reply:
I didn't realise we had that on the channel. What I would prefer actually is to change both the channel and the thread to have a `last_message_id` . In the case of the channel it is the pointer to the last message sent which can get the date + time etc as needed, and in the case of threads it points to the last reply which is used in the thread list and for thread indicators, as well as for tracking.

> just use a tracking state report when we need it in a serializer

IMO in the serializer we would change it to have a `has_one` like this:

```ruby
has_one :last_message, serializer: Chat::MessageSerializer, embed: :objects
```

In the case of threads, we already serialize the last reply via `ThreadPreviewSerializer`

The reason I have aded it to the tracking state here is because I need this value alongside the unread state for every thread across the channels to be able to show the blue dot. We don't need to use the tracking state just for regular channel serialization just for this last message thing.

-----

So, perhaps we can step back here, and I will go add `last_message_id` on the channel and thread DB tables in another PR before continuing with this one?

Reply:
This is the same column discussed in /t/105035 internally, but that was only threads. So saying here we should have it for channel too. They are both easy enough to keep in sync because we have message create/trash services.

Reply:
What happens with these ids when we delete a message?

Reply:
We have the TrashMessage service, we can just update them to the new last ID, that's what I was saying here:

> They are both easy enough to keep in sync because we have message create/trash services.

Reply:
@jjaffeux so is this fine to do, you think? I want to do this plan before continuing this PR in a separate PR:

1. Add `last_message_id` to `Chat::Channel` and `Chat::Thread` models/table
2. Update thread queries looking for `last_reply` to look at `last_message_id`. This should speed up the index and indicators for threads
3. Replace use of `last_message_sent_at` for channel to join on `last_message_id` and get the `created_at` from there
4. Update the `ChatMessageCreator` and `TrashMessage` services to update the `last_message_id` for channel and thread

This should clean up a lot of things and make it more consistent.

Reply:
Yes fine with this :+1: I think this is good changes